src/app.rs

```diff
@@ -299,6 +299,20 @@ impl Config {
     pub fn get_repoconfig(&self, name: &str) -> Result<&RepoConfig, ApiError> {
         self.repos.get(name).ok_or_else (|| ApiError::BadRequest("No such repo".to_string()))
     }
+
+    pub fn get_repoconfig_from_path(&self, path: &Path) -> Result<&RepoConfig, ApiError> {
+        let mut buf = PathBuf::new();
+
+        for part in path.iter() {
+            buf.push(part);
+            if let Some(name) = buf.to_str() {
+                if let Some(config) = self.repos.get(name) {
+                    return Ok(config)
+                }
+            }
+        }
+        Err(ApiError::BadRequest("No such repo".to_string()))
+    }
```

Comment:
I still think this is a bit overly complex, couldn't you just do 

```
for (prefix, config) in self.repos.iter() {
   if path.starts_with(prefix) {
     return Ok(config)
  } 
}
```

Reply:
Right. You'd still need to be careful to differentiate between a repo named `foo` and one named `foobar`. I'll try again.

Reply:
From Path.starts_with() docs: Only considers whole path components to match.

Reply:
Oh, missed that was an actual path method. I thought you were suggesting a strong comparison.